### PR TITLE
Allows cond_v2 to return tensors in nested output structures. Resolved merge conflicts and squashed into one commit.

### PR DESCRIPTION
**Pull request by pythonanonuser**
@skye Had to reopen the pull request since it got closed for some reason while I was squashing commits

### CODE REVIEW

File: tensorflow/python/ops/cond_v2.py [hunk 1]
```diff
@@ -136,11 +136,11 @@ def cond_v2(pred, true_fn, false_fn, name="cond"):
...
 
-    result = tuple(tensors[:num_cond_outputs])
-    if len(result) == 1:
-      return result[0]
-    else:
-      return result
+    # Packing output tensors in the same nested structure as the true and false
+    # functions return
+    result = nest.pack_sequence_as(structure=true_graph.structured_outputs, 
+                                   flat_sequence=tensors[:num_cond_outputs])
```
Comment (skye) [inline tensorflow/python/ops/cond_v2.py]: Didn't you have a check in _check_same_outputs that true_graph and false_graph have the same outputs?

**Comment by pythonanonuser** [comment]
@skye My apologies there was another versioning error on my end. Should be fixed with the latest commit.

File: tensorflow/python/ops/cond_v2.py (hunk 1 above)
Comment (pythonanonuser) [reply tensorflow/python/ops/cond_v2.py]: @skye My apologies there was another versioning error on my end. Should be fixed with the latest commit.

**Review by skye — DISMISSED** [review]

**Comment by skye** [comment]
@ymodak can you rerun the tests and try to get this in? I'm OOO until Tuesday (and can't find how to add labels on my phone)

**Comment by ymodak** [comment]
@skye  Yes. I will work on this.

**Comment by pythonanonuser** [comment]
@skye @ymodak Sorry there was a minor typo when copying the last fix. All tests should pass now.

**Pull request merged**